tests/Unit/Support/Constraint/Fixtures/TestingCssConstraint.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Pelago\Emogrifier\Tests\Unit\Support\Constraint\Fixtures;
+
+use Pelago\Emogrifier\Tests\Support\Constraint\CssConstraint;
+
+/**
+ * Extends the `CssConstraint` class to provide indirect access to protected method(s) for testing.
+ *
+ * @author Jake Hotson <jake.github@qzdesign.co.uk>
+ */
+abstract class TestingCssConstraint extends CssConstraint
```

Comment:
If I read this correctly, this class is intended to be neither sublassed nor instantiated. In this case, it should be `abstract final` or `final abstract` (I don't know if PHP allows this, though).

Reply:
A class cannot be both `abstract` and `final`.  So I have not changed anything here.  If you want, I could expand on the commentary, but OTOH, who knows what the future holds?  It's not intended to be subclassed or instantiated, but there might be some reason to do so that we haven't thought of yet.